src/components/TodoList/TodoList.tsx

```diff
@@ -0,0 +1,49 @@
+import { FC, memo } from 'react';
+import { TodoItem } from '../TodoItem/TodoItem';
+import { Todo } from '../../types/Todo';
+
+export interface TodoListProps {
+  todos: Todo[],
+  onDeleteTodo: (todoId: number) => void;
+  tempTodo: Todo | null,
+  deletingTodoIds: number[],
+  updateTodo:(
+    todoId: number,
+    fieldsToUpdate: Partial<Pick<Todo, 'title' | 'completed'>>
+  ) => Promise<void>,
+  updatingTodosIds: number[],
+}
+
+export const TodoList: FC<TodoListProps> = memo(({
+  todos,
+  onDeleteTodo,
+  tempTodo,
+  deletingTodoIds,
+  updateTodo,
+  updatingTodosIds,
+}) => {
+  return (
+    <section className="todoapp__main" data-cy="TodoList">
+      {todos.map(todo => (
+        <TodoItem
+          key={`${todo.id}`}
```

Comment:
```suggestion
          key={todo.id}
```
No need to convert key into string. Just pass the todo.id